clean/new_orleans_pd_uof.py

```diff
@@ -0,0 +1,732 @@
+from lib.columns import clean_column_names
+import deba
+from lib.clean import clean_dates, standardize_desc_cols
+from lib.columns import set_values
+from lib.uid import gen_uid
+import pandas as pd
+import sys
+
+sys.path.append("../")
+
+
+def join_citizen_columns(df):
+    df.loc[:, "citizen"] = (
+        df.subject_ethnicity.fillna("null")
+        + " ; "
+        + df.subject_gender.fillna("null")
+        + " ; "
+        + df.subject_hospitalized.fillna("null")
+        + " ; "
+        + df.subject_injured.fillna("null")
+        + " ; "
+        + df.subject_influencing_factors.fillna("null")
+        + " ; "
+        + df.subject_distance_from_officer.fillna("null")
+        + " ; "
+        + df.subject_age.fillna("null")
+        + " ; "
+        + df.subject_build.fillna("null")
+        + " ; "
+        + df.subject_height.fillna("null")
+        + " ; "
+        + df.subject_arrested.fillna("null")
+        + " ; "
+        + df.subject_arrest_charges.fillna("null")
+    )
+    df.loc[:, "citizen"] = (
+        df.citizen.str.lower()
+        .str.strip()
+        .fillna("null")
+        .astype(str)
+        .str.replace(r">", "", regex=True)
+        .str.replace(r"<", "", regex=True)
+        .str.replace(r'"', "", regex=True)
+        .str.replace(r"'", "", regex=True)
+        .str.replace(r"- ?", "", regex=True)
+        .str.replace(r"\|  \|", "| null |", regex=True)
+        .str.replace(r"\|  \)", "| null)", regex=True)
+        .str.replace(
+            r"(\w+) (\w+) ?(\w+)? ?(\w+)? ?(\w+)? ?(\w+)?",
+            r"\1\2\3\4\5\6",
+            regex=True,
+        )
+        .str.replace(r"(\w+)\|", r"\1 |", regex=True)
+        .str.replace(r"(\w+);", r"\1 ;", regex=True)
+        .str.replace(
+            r"(\w+) \| (\w+) ; (\w+) \| (\w+) ; (\w+) \| (\w+) ; (\w+) \| (\w+) ; "
+            r"(\w+) \| (\w+) ; (\w+) \| (\w+) ; (\w+) \| (\w+) ; (\w+) \| (\w+) ; "
+            r"(\w+) \| (\w+) ; (\w+) \| (\w+) ; (\w+) ?\|? ?(\w+)?",
+            r"(\1 | \3 | \5 | \7 | \9 | \11 | \13 | \15 | \17 | \19 | \21) & "
+            r"(\2 | \4 | \6 | \8 | \10 | \12 | \14 | \16 | \18 | \20 | \22)",
+            regex=True,
+        )
```

Comment:
This is very hard to understand. And regular expression is actually very slow compared to other string manipulation methods. I think you should rewrite `join_citizen_columns` to not rely on regular expression so much anymore.

Reply:
Ok.